src/record_syscall.cc

```diff
@@ -6393,8 +6393,20 @@ static void rec_process_syscall_arch(RecordTask* t,
   // syscall completes --- and that our TaskSyscallState infrastructure can't
   // handle.
   switch (syscallno) {
+    case Arch::vfork: {
+      if (t->emulated_ptrace_options & PTRACE_O_TRACEVFORKDONE) {
+        t->emulate_ptrace_stop(
+            WaitStatus::for_ptrace_event(PTRACE_EVENT_VFORK_DONE));
+      }
+      // see Arch::clone, Arch::vfork used to fall-through there.
```

Comment:
Why can't it continue to fall through here?

Reply:
I think you told me to separate it (or maybe I had misunderstood)? I had initially the `[[fallthrough]]` attribute there, so I had thought maybe that was not according to convention.

I think the `[[fallthrough]]` attribute is ok, in c++14, right? That way we won't get any warnings about it.

Reply:
Yeah I don't think I understood how simple this was when I told you that :)

We use RR_FALLTHROUGH IIRC.

Reply:
Great :)